src/test/java/com/shapesecurity/shift/validator/ValidatorTest.java

```diff
@@ -0,0 +1,536 @@
+    assertCorrectFailures(test2, 1, ValidationErrorMessages.VALID_IF_STATEMENT);
+  }
+
+  @Test
+  public void testImportSpecifier() {
+    Module test0 = new Module(ImmutableList.nil(), ImmutableList.list(new Import(Maybe.just(new BindingIdentifier("a")), ImmutableList.list(new ImportSpecifier(Maybe.nothing(), new BindingIdentifier("b"))), "c")));
+    assertCorrectFailures(test0, 0, "");
+
+    Module test1 = new Module(ImmutableList.nil(), ImmutableList.list(new Import(Maybe.just(new BindingIdentifier("a")), ImmutableList.list(new ImportSpecifier(Maybe.just("a"), new BindingIdentifier("b"))), "c")));
+    assertCorrectFailures(test1, 0, "");
+
+    Module test2 = new Module(ImmutableList.nil(), ImmutableList.list(new Import(Maybe.just(new BindingIdentifier("a")), ImmutableList.list(new ImportSpecifier(Maybe.just("2d849"), new BindingIdentifier("b"))), "c")));
+    assertCorrectFailures(test2, 1, ValidationErrorMessages.VALID_IMPORT_SPECIFIER_NAME);
+  }
+
+  @Test
+  public void testLabeledStatement() {
+    Script test0 = new Script(ImmutableList.nil(), ImmutableList.list(new LabeledStatement("done", new WhileStatement(new LiteralBooleanExpression(true), new BlockStatement(new Block(ImmutableList.list(new BreakStatement(Maybe.just("done")))))))));
+    assertCorrectFailures(test0, 0, "");
+
+    Script test1 = new Script(ImmutableList.nil(), ImmutableList.list(new LabeledStatement("5346done", new WhileStatement(new LiteralBooleanExpression(true), new BlockStatement(new Block(ImmutableList.list(new BreakStatement(Maybe.just("done")))))))));
+    assertCorrectFailures(test1, 1, ValidationErrorMessages.VALID_LABEL);
+
+    Script test2 = new Script(ImmutableList.nil(), ImmutableList.list(new LabeledStatement("#das*($839da", new WhileStatement(new LiteralBooleanExpression(true), new BlockStatement(new Block(ImmutableList.list(new ContinueStatement(Maybe.just("done")))))))));
+    assertCorrectFailures(test2, 1, ValidationErrorMessages.VALID_LABEL);
+
+  }
+
+  @Test
+  public void testLiteralNumericExpression() throws JsError {
+    Script test0 = Parser.parseScript("1.0");
+    assertCorrectFailures(test0, 0, "");
+
+    Script test1 = new Script(ImmutableList.nil(), ImmutableList.list(new ExpressionStatement(new LiteralNumericExpression(-1.0))));
+    assertCorrectFailures(test1, 1, ValidationErrorMessages.LITERAL_NUMERIC_VALUE_NOT_NEGATIVE);
+
+    Script test2 = new Script(ImmutableList.nil(), ImmutableList.list(new ExpressionStatement(new LiteralNumericExpression(1.0/0))));
+    assertCorrectFailures(test2, 1, ValidationErrorMessages.LITERAL_NUMERIC_VALUE_NOT_INFINITE);
+
+    Script test3 = new Script(ImmutableList.nil(), ImmutableList.list(new ExpressionStatement(new LiteralNumericExpression(Double.NaN))));
+    assertCorrectFailures(test3, 1, ValidationErrorMessages.LITERAL_NUMERIC_VALUE_NOT_NAN);
+
+  }
+
+  @Test
+  public void testLiteralRegExpExpression() {
```

Comment:
Please make sure each test has at least one example without a failure. This test does not have one.